doc/source/handling-dependencies.rst

```diff
@@ -0,0 +1,216 @@
+.. _handling_dependencies:
+
+Handling Dependencies
+=====================
+
+Your Ray application may depend on environment variables, files, and Python packages.
+Ray provides two features to specify these dependencies when working with a remote cluster: Runtime environments, and the Ray cluster launcher commands
+With these features, you no longer need to manually SSH into your cluster and set up your environment.
+
+**Option 1.**  You can specify dependencies dynamically at runtime in Python using :ref:`Runtime Environments<runtime-environments>`, described below.  
+This can be useful for
+
+ - quickly iterating on a project with changing dependencies and files, and 
+ 
+ - for running jobs, tasks and actors with different dependencies, all on the same Ray cluster.
+
+**Option 2.**  Alternatively, you can prepare your Ray cluster's environment when your cluster nodes start up, and modify it later from the command line.
+Packages can be installed using ``setup_commands`` in the Ray Cluster configuration file (:ref:`docs<cluster-configuration-setup-commands>`) and files can be pushed to the cluster using ``ray rsync_up`` (:ref:`docs<ray-rsync>`).
```

Comment:
@rkooo567 I know we still need more here but I'm not quite sure what to put, do you have any ideas?

Reply:
I think we should ask the autoscaler team to fill it up.

Reply:
I think common problems are 

Manual:
- Link to autoscaler section that describes how to set up deps
- env variables (setup commands)
- System deps (setup commands)
- Files (rsync up or manually copy and paste. Make sure they are all synced)
- Python packages (setup commands)

Container
- Same things (link to container deployment)